app/core/utils.coffee

```diff
@@ -61,9 +61,14 @@ translatejs2cpp = (jsCode, fullCode=true) ->
     jsCodes[i] = jsCodes[i].replace new RegExp(' !== ', 'g'), ' != '
     jsCodes[i] = jsCodes[i].replace new RegExp(' var ', 'g'), ' auto '
     jsCodes[i] = jsCodes[i].replace new RegExp(' = \\[(.*)\\]', 'g'), ' = {$1}'
+    jsCodes[i] = jsCodes[i].replace new RegExp('\\(var ', 'g'), '(auto '
```

Comment:
If we wanted to make this cleaner, I think there's no reason to use `new RegExp` constructor over the `//` RegExp literal syntax unless we're interpolating a variable, so these escapes (plus the existing ones above) could go away:

```suggestion
    jsCodes[i] = jsCodes[i].replace /\(var /g, '(auto ')
```

Reply:
Another minor cleanup if we're at it: how about indenting the `return 0;` on line 45? It seems odd to me that there's no indentation on that line. Maybe that's the style? We end up with C++ sample code like this:

```cpp
int main {
    hero.moveRight();

return 0;
}
```

whereas I would think it'd be like this:

```cpp
int main {
    hero.moveRight();

    return 0;
}
```

Reply:
yeah, actually why i use `new RegExp` is `//` in coffee has little different with js? i write `//` in js and copy to coffee didn't work. i think it is about escape characters.. i will update this.

Reply:
Should be the same as in JS, apart from additional support for [block regular expressions](http://coffeescript.org/#regexes).